### ISSUE TITLE
Snowflake VARCHAR of max length no longer matches DESCRIBE

### ISSUE THREAD

**Issue by bchin3** [littleK0i/SnowDDL#22]
A recent Snowflake update appears to no longer output the length of a VARCHAR field when it is max length.  So "VARCHAR(16777216)" now appears as simply "VARCHAR".  This causes SnowDDL to return errors and not execute.  2 cases below

DESCRIBE now returns this:
```
name	type	kind	null?	default	primary key	unique key	check	expression	comment	policy name
ACTOR_ID	NUMBER(10,0)	COLUMN	N		N	N				
FIRST_NAME	VARCHAR(45)	COLUMN	N		N	N				
LAST_NAME	VARCHAR	COLUMN	Y		N	N	
```		

1) When specifying the max length in the table.yaml file

```
columns:
  actor_id: NUMBER(10,0) NOT NULL
  first_name: VARCHAR(50) NOT NULL
  last_name: VARCHAR(16777216)

```
I receive this error:

```
  Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\envs\snowflake\lib\site-packages\snowddl\resolver\abc_resolver.py", line 115, in _process_tasks
    result = f.result()
  File "C:\ProgramData\Anaconda3\envs\snowflake\lib\concurrent\futures\_base.py", line 437, in result
    return self.__get_result()
  File "C:\ProgramData\Anaconda3\envs\snowflake\lib\concurrent\futures\_base.py", line 389, in __get_result
    raise self._exception
  File "C:\ProgramData\Anaconda3\envs\snowflake\lib\concurrent\futures\thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "C:\ProgramData\Anaconda3\envs\snowflake\lib\site-packages\snowddl\resolver\table.py", line 62, in compare_object
    snow_cols = self._get_existing_columns(bp)
  File "C:\ProgramData\Anaconda3\envs\snowflake\lib\site-packages\snowddl\resolver\table.py", line 267, in _get_existing_columns
    type=DataType(dtype),
  File "C:\ProgramData\Anaconda3\envs\snowflake\lib\site-packages\snowddl\blueprint\data_type.py", line 98, in __init__
    self.val1 = int(m['val1']) if self.base_type.number_of_properties >= 1 else None
TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
```

2) When specifying no length in the table.yaml file

```
columns:
  actor_id: NUMBER(10,0) NOT NULL
  first_name: VARCHAR(50) NOT NULL
  last_name: VARCHAR
```

I receive this error:

```
Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\envs\snowflake\lib\site-packages\snowddl\parser\abc_parser.py", line 38, in parse_schema_object_files
    callback(file)
  File "C:\ProgramData\Anaconda3\envs\snowflake\lib\site-packages\snowddl\parser\table.py", line 170, in process_table
    type=DataType(m.group('type')),
  File "C:\ProgramData\Anaconda3\envs\snowflake\lib\site-packages\snowddl\blueprint\data_type.py", line 98, in __init__
    self.val1 = int(m['val1']) if self.base_type.number_of_properties >= 1 else None
TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
```

**Comment by littleK0i** [comment]
Yes, I've discussed it with Snowflake Customer Support.

They agree it is a "regression" introduced in Snowflake version `6.32.0`, and it should be fixed in the next version.

In the meantime you may ask customer support to revert your account back to `6.31.*`. They can do it very fast.

Adding workaround in code is a bit complicated. Maybe it is not worth it for just one version which will be phased out soon.

**Comment by bchin3** [comment]
Snowflake Support has confirmed that this is an issue cause by their recent 6.32 release and should be corrected in their 6.33 release this week.  I suggest to leave this issue open until we can confirm that the new Snowflake release indeed resolves this for all accounts.

**Comment by littleK0i** [comment]
I see the problem was fixed in `6.32.3`. This is no longer an issue.

**Comment by bchin3** [comment]
Yes, seems to be fixed in 6.32.3
